### PR TITLE
feat: track chats

### PR DESCRIPTION
**Pull request by roziscoding**
Solves #3

### CODE REVIEW

File: src/mod.ts [hunk 1]
```diff
@@ -139,4 +141,110 @@ export function chatMembers(
...
+    getChat: (chatId?: number) => Promise<Chat>;
+  };
+};
+
+type ChatsContext = Context & ChatsFlavor;
+
+type ChatsOptions = {
+  /**
+   * Prevents deletion of chats when
+   * bot is blocked or kiecked.
```
Comment (roziscoding) [inline src/mod.ts]:
```suggestion
   * bot is blocked or kicked.
```

File: src/mod.ts [hunk 2]
```diff
@@ -139,4 +141,110 @@ export function chatMembers(
...
+   * then proceeds to call `ctx.chatMembers.getChat` to add the chat information to the storage in case it
+   * doesn't exist.
+   *
+   * Please note that this means the storage will be called for **every update**, which may be a lot, depending on how many
+   * updates your bot receives. This also has the potential to impact the performance of your bot drastically. Only use this
+   * if you _really_ know what you're doing and are ok with the risks and consequences.
+   */
+  enableAggressiveStorage: boolean;
+  /**
+   * Controls wether the middleware should should also store private chats
```
Comment (roziscoding) [inline src/mod.ts]:
```suggestion
   * Controls wether the middleware should also store private chats
```

**Review by mrmaster009 — COMMENTED** [review]
Typo on line 182 - wether -> whether
                        193 - <YOR_TELEGRAM_TOKEN> -> <YOUR_TELEGRAM_TOKEN>

**Comment by roziscoding** [comment]
Closing in favor of #6

**Pull request closed without merging**